### DISCUSSION TITLE
Binary protocol

### DISCUSSION THREAD

**Discussion by uasan** [porsager/postgres#258] [Ideas]
I know the @porsager  is interested in supporting the binary protocol, so I created this thread.

There are several advantages to a binary protocol:
 1. Binary protocol, gives better typing to nested structures, arrays and records inside store oid types of elements, you do not need to parse a string
 2.  Less bytes for some data types, for example, uuid - 16 bytes instead of 36 in text form, timestamp - 8 bytes instead of 24 in text form
 3. Better performance, because postgres gives data in the form in which it is stored
 4. Zero cost for bytea type

I advise you to study the experience of other clients

C# client: https://github.com/npgsql/npgsql/issues/447
Description binary struct by type: https://github.com/npgsql/doc/blob/main/dev/types.md/#L1

I also develop a client on node.js, later I can lay out a draft version

**Comment by unilynx** [comment]
I'm using https://github.com/panates/postgrejs and it has already implemented a binary protocol. Which means they've already built all the data type encoders - https://github.com/panates/postgrejs/tree/main/src/data-types - under a MIT license which I assume is compatible

I'm no PG protocol export, but I still see "SELECT" statements going over the wire with postgrejs - is it just the actual data (parameters and rows) that differ in binary/text format or is there a lot more to it ?
